biz.aQute.repository/test/aQute/bnd/repository/p2/provider/P2RepositoryTest.java

```diff
@@ -36,4 +42,21 @@ public void testSimple() throws Exception {
 		}
 	}
 
+	// @Test does not work for some reason
+	public void testXtext() throws Exception {
```

Comment:
So this is a test which is not executed?

Reply:
Thanks for keeping me honest. There was actually quite a fundamental flaw in the P2Impl ... It resolved against the base repo, not against the repo that contained the compositeArtifacts.xml we were parsing. A lot of complex P2 repos were not working, only simple P2 repos worked well.